File: app/src/main/java/com/liang/jtablayout/MainActivity.java
package com.liang.jtablayout;

import android.content.Intent;
import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;

import com.liang.widget.BadgeView;

public class MainActivity extends AppCompatActivity {

    BadgeView badgeView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);

        findViewById(R.id.button).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, TabActivity.class));
            }
        });

        findViewById(R.id.button1).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, OperationActivity.class));
            }
        });

        findViewById(R.id.button2).setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this, Tab2Activity.class));
            }
        });


        badgeView = findViewById(R.id.badgeView);
        badgeView.show("HOT");

    }
}

File: tab3xdemo/src/main/java/com/liang/tab3xdemo/MainActivity.kt
package com.liang.tab3xdemo

import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import com.liang.tablayout3x.GridLayoutManager
import com.liang.tablayout3x.LinearLayoutManager
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
//        tabLayout.layoutManager = LinearLayoutManager(this)
//        repeat(20) {
//            tabLayout.addTab(tabLayout.newTab().setTitle("Test: $it"))
//        }
    }
}
